backend/substrapp/tasks/tasks.py

```diff
@@ -487,10 +487,15 @@ def prepare_tuple(subtuple, tuple_type):
     # Can happen if we re-process all events
     # But we need to fetch the subtuple again to get the last
     # version of it
-    _, subtuple_check = find_training_step_tuple_from_key(subtuple['key'])
-    if subtuple_check['status'] != 'todo':
-        logger.error(f'Tuple task ({tuple_type}) not in "todo" state ({subtuple_check["status"]}).\n{subtuple_check}')
-        return
+    try:
+        _, subtuple_check = find_training_step_tuple_from_key(subtuple['key'])
+        if subtuple_check['status'] != 'todo':
+            logger.error(f'Tuple task ({tuple_type}) not in "todo" state ({subtuple_check["status"]}).'
+                         f'\n{subtuple_check}')
+            return
+    except Exception:
+        # use the provided subtuple if the previous call fail
+        pass
```

Comment:
does the provided subtuple have a status ? Or is it not reliable to check the `subtuple['status'] == 'todo'` ?

Reply:
Subtuple provide a status. It's not always reliable in case of backend-server restart and replaying events from the begining. Indeed, a replayed event gives the worker a subtuple from an old state of the ledger. This can be seen as a sanity check for replayed event. New events do not need this sanity check :)